exonum/src/runtime/mod.rs

```diff
@@ -57,24 +59,30 @@ pub enum ArtifactSpec {
 /// Service runtime environment.
 /// It does not assign id to services/interfaces, ids are given to runtime from outside.
 pub trait RuntimeEnvironment {
-    /// Deploy artifact.
-    fn deploy(&self, artifact: ArtifactSpec) -> Result<(), DeployError>;
+    /// Start artifact deploy.
+    fn start_deploy(&self, artifact: ArtifactSpec) -> Result<(), DeployError>;
+
+    /// Check deployment status.
+    fn check_deploy_status(&self, artifact: ArtifactSpec) -> Result<DeployStatus, DeployError>;
```

Comment:
I find this result to be a little bit redundant since `DeployStatus` contain `DeployError` variant.

Reply:
Ah, I forgot to remove this variant. I decided that using a `Result` is a bit more convenient than implementing it inside of enum as variant. I'll remove `DeployError` status.